src/govuk/components/tabs/tabs.mjs

```diff
@@ -57,29 +83,42 @@ Tabs.prototype.setup = function () {
     $item.setAttribute('role', 'presentation')
   })
 
-  nodeListForEach($tabs, function ($tab) {
-    // Set HTML attributes
-    this.setAttributes($tab)
+  nodeListForEach(
+    $tabs,
 
-    // Save bounded functions to use when removing event listeners during teardown
-    $tab.boundTabClick = this.onTabClick.bind(this)
-    $tab.boundTabKeydown = this.onTabKeydown.bind(this)
+    /**
+     * Loop through tab links
+     *
+     * @param {HTMLAnchorElement} $tab - Tab link
+     * @this {Tabs}
+     */
+    function ($tab) {
+      // Set HTML attributes
+      this.setAttributes($tab)
 
-    // Handle events
-    $tab.addEventListener('click', $tab.boundTabClick, true)
-    $tab.addEventListener('keydown', $tab.boundTabKeydown, true)
+      // Save bounded functions to use when removing event listeners during teardown
+      $tab.boundTabClick = this.onTabClick.bind(this)
+      $tab.boundTabKeydown = this.onTabKeydown.bind(this)
```

Comment:
**suggestion**: Rather than poluting the element, these can be stored on `this` as it'll also be available when removing the listener. This'll also save having one bound function per tab, as all the component really needs is one bound function for the whole component (those could be created in the `init`. This would fix a few extra typing issues.

Reply:
Glad this caught your eye too

It's one of the few areas that fail the `tsc --build` checks in:

* https://github.com/alphagov/govuk-frontend/pull/2971

It's a technique used in the GOV.UK Publishing components:
[govuk_publishing_components/components/modal-dialogue.js#L20](https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/javascripts/govuk_publishing_components/components/modal-dialogue.js#L20)

I'll update it tomorrow

Reply:
Resolved in commit **"Move Tab bound handlers to constructor"**